src/Orbit/main.cpp

```diff
@@ -255,6 +256,59 @@ static void LogAndMaybeWarnAboutClockResolution() {
   }
 }
 
+static bool IsDirectoryEmpty(const std::filesystem::path& directory) {
+  auto exists_or_error = orbit_base::FileExists(directory);
+  if (exists_or_error.has_error()) {
+    ERROR("Unable to stat \"%s\": %s", directory.string(), exists_or_error.error().message());
+    return false;
+  }
+
+  if (!exists_or_error.value()) {
+    return true;
+  }
+
+  auto file_list_or_error = orbit_base::ListFilesInDirectory(directory);
+  if (file_list_or_error.has_error()) {
+    ERROR("Unable to list directory \"%s\": %s", directory.string(),
+          file_list_or_error.error().message());
+    return false;
+  }
+
+  return file_list_or_error.value().empty();
+}
+
+static void TryMoveSavedDataLocationIfNeeded() {
+  if (IsDirectoryEmpty(orbit_core::GetPresetDirPriorTo1_65()) &&
+      IsDirectoryEmpty(orbit_core::GetCaptureDirPriorTo1_65())) {
+    return;
+  }
+
+  auto dialog = orbit_qt::MoveFilesDialog::Create();
+  orbit_qt::MoveFilesProcess process;
+
+  QObject::connect(&process, &orbit_qt::MoveFilesProcess::generalError,
+                   [&dialog](const std::string& error_message) {
+                     dialog->AddText(absl::StrFormat("Error: %s", error_message));
+                   });
+
+  QObject::connect(
+      &process, &orbit_qt::MoveFilesProcess::moveStarted,
+      [&dialog](const std::filesystem::path& from_dir, const std::filesystem::path& to_dir,
+                size_t number_of_files) {
+        dialog->AddText(absl::StrFormat(R"(Moving %d files from "%s" to "%s" ...)", number_of_files,
+                                        from_dir.string(), to_dir.string()));
+      });
+
+  QObject::connect(&process, &orbit_qt::MoveFilesProcess::moveDone,
+                   [&dialog]() { dialog->AddText("... done"); });
+
+  QObject::connect(&process, &orbit_qt::MoveFilesProcess::processFinished,
+                   [&dialog]() { dialog->EnableCloseButton(); });
+
+  process.Run();
+  dialog->exec();
```

Comment:
What happens when the user presses Escape? Will this close the dialog while moving files is still in progress? I would assume that the dialog closes, `process`'s destructor will block in that case and Orbit's UI will be frozen until moving files is done.

Reply:
this is a bit hard to test - the process is very fast since when we move on the same device it effectively renames. I will try to insert some sleeps and see what happens. I expect the same. Btw UI is not yet up at this point so the user will see nothing until the process is done.

Reply:
Should we do something about it? We could override the `closeEvent` method (https://doc.qt.io/qt-5/qwidget.html#closeEvent) and ignore the event if we aren't done yet.

Reply:
I must be doing something wrong. closeEvent is not getting triggered on escape.